hibernate-core/src/main/java/org/hibernate/sql/InsertSelect.java

```diff
@@ -45,7 +45,7 @@ public InsertSelect addColumn(String columnName) {
 	}
 
 	public InsertSelect addColumns(String[] columnNames) {
-		this.columnNames.addAll(Arrays.asList(columnNames));
+		Collections.addAll( this.columnNames,columnNames );
```

Comment:
You could additionally change the type of `columnNames` to `ArrayList` and use `ensureCapacity` before adding.